dali/pipeline/data/tensor.h

```diff
@@ -218,6 +218,37 @@ class Tensor : public Buffer<Backend> {
     shares_data_ = num_bytes_ > 0 ? true : false;
   }
 
+  /**
+   * @brief Wraps the raw allocation. The input pointer must not be nullptr.
+   * if the size of the allocation is zero, the Tensor is reset to a default
+   * state and is NOT marked as sharing data. Also sets shape of new Tensor.
+   *
+   * After wrapping the allocation, the Tensors size is set to dot product
+   * of shape vector, and its type is reset to NoType. Future calls to
+   * Resize or setting of the Tensor type will evaluate whether or not the
+   * current allocation is large enough to be used and proceed appropriately.
+   *
+   * The Tensor object assumes no ownership of the input allocation, and will
+   * not de-allocate it when it is done using it. It is up to the user to
+   * manage the lifetime of the allocation such that it persist while it is
+   * in use by the Tensor.
+   */
+  inline void ShareData(void *ptr, size_t bytes, const vector<Index> &shape) {
+    DALI_ENFORCE(ptr != nullptr, "Input pointer must not be nullptr.");
+
+    // Save our new pointer and bytes. Reset our type, shape, and size
+    data_.reset(ptr, [](void *) {});
```

Comment:
curious, why do we need a smart pointer if we don't use the deleter?

Reply:
We use - please look into ResizeHelper.